README.md

```diff
@@ -28,22 +61,103 @@ mtx.myTranspose();
-8. Use `myMap` to build a new array where every non-string element is converted to an empty string. 
+
+console.log(arr.myMap((el)=>{
+  return el + 10
+}))
+
+8. Use `myMap` to build a new array where every non-string element is converted to an empty string.
+
+let test = [1,2,3,4,"a","b","c"]
+
+test.myMap((el)=>{
+  if(typeof(el) !== "string") {
+    return("")}
+      else{
+        return(el)}
+});
+
 9. Use `myFilter` to get only the even elements in an array.
-10. Use `myEvery` to check if all elements in the array are the same. 
-11. Use `myReduce` to return the sum of every element in an array. 
-12. Use `myMap` to build a new array that doubles each element. Then chain `myReduce` to find the product of all the elements. 
-13. Use `myReduce` to find the largest number in an array. 
+
+arr.myFilter((el)=>{
+  return el%2 === 0
+})
+
+10. Use `myEvery` to check if all elements in the array are the same.
+
+// because we're testing to see if all the elements to see if they're the same it should be the same as just testing the first //element.
+
+let arr = [1,2,3,4]
+
+arr.myEvery((el)=>{
+  return el === this[0]
+});
+
+//results in false
+
+11. Use `myReduce` to return the sum of every element in an array.
+
+
+12. Use `myMap` to build a new array that doubles each element. Then chain `myReduce` to find the product of all the elements.
+
+13. Use `myReduce` to find the largest number in an array.
 14. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.
 Use `myFilter` to return a new array of every element of arr that can be evenly divided by divisor.
-14. Write a `myJoin` function. 
+14. Write a `myJoin` function.
+
+Array.prototype.myJoin = function(callback){
+  let joinedStr = ""
+  for (let i=0;i<this.length;i++){
+    if (this[i]){
+      joinedStr+=this[i]
+    }
+  }return joinedStr
+}
+
+let exp = ["banana","","words",""]
+exp.myJoin()
+
 15. Write a `mySlice` function.
+
+let str = "this is a lot of random words"
+
+// because slice can work without a second argument we have to set a default to the end of the string
+String.prototype.mySlice = function(ar1,ar2=this.length-1){
+  let sliced = '';
+  for(let i=ar1;i<ar2;i++){
+    sliced += str[i]
+  }return sliced
+}
+
+str.mySlice(3,10)
+
 16. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
-Use reduce.
-17. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by it's index. 
 
-Bonus: Write a `myFlatten`. This should take a multi-dimensional array and return it as one array. 
+function countZeroes (el){
+  let count = 0
+  for (let i=0; i<arr.length;i++){
+    if (arr[i] === 0){
+      count++
+    }
+  }return count
+};
+
+
+17. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by it's index.
+
+let test = [1,2,3,4,5]
+// unsure how to use map in this situation
```

Comment:
This description is good - but it describes what map is already doing for us under the hood! I think it'd be worthwhile to try to use myMap and see what you get.